### PR TITLE
Add iterative templates.

### PR DESCRIPTION
**Pull request by tecimovic**
This is addressing: 
https://github.com/project-chip/zap/issues/1042

Iterative templates, are the template files that have a name like something_{id}.out, for example,
and a property "iterator" set on the individual file.

Depending on iterator (for example "available_clusters") the generator will iterate over all the 
values of an interator, and generate a separate file for each one, substituting {id} with a given value.

There will be different values for "iterator" and for the {pattern} thingies in the output file names.

### CODE REVIEW

File: src-electron/db/query-endpoint-type.js [hunk 1]
```diff
@@ -165,14 +165,16 @@ async function selectEndpointType(db, id) {
 async function selectAllClustersDetailsFromEndpointTypes(
   db,
   endpointTypes,
-  options = null
```
Comment (brdandu) [inline src-electron/db/query-endpoint-type.js:168]: Passing options allows other flags coming from the handlebar helper. This helps in not updating the function signature.

**Review by brdandu — APPROVED** [review]

**Review by paulr34 — APPROVED** [review]

File: src-electron/db/query-endpoint-type.js (hunk 1 above)
Comment (tecimovic) [reply src-electron/db/query-endpoint-type.js:168]: The intent is valid. But the options were passed in a handlebars specific way (options.hash) and I needed to call this functions out of template context. I don't want db layer to be polluted by handlebars specifics.

File: test/resource/meta/gen-test.json [hunk 2]
```diff
@@ -33,6 +37,12 @@
       "path": "access.zapt",
       "name": "Access data example",
       "output": "access.out"
+    },
+    {
+      "path": "cluster_iterator.zapt",
+      "name": "Cluster iterator.",
+      "output": "cluster_out_{define:tolowercase}.out",
+      "iterator": "selectedCluster"
```
Comment (bzbarsky-apple) [inline test/resource/meta/gen-test.json:45]: How does one discover valid "iterator" values, @tecimovic ?  The docs talk about "availableClusters" as a value, but dbenum does not seem to match that, and nothing short of reading the code will discover "selectedCluster"....

Comment (tecimovic) [reply test/resource/meta/gen-test.json:45]: I need to update sdk-integration.md doc....

**Pull request merged**